aten/src/ATen/native/TensorConversions.cpp

```diff
@@ -756,9 +785,101 @@ Tensor dense_to_sparse_bsr(const Tensor& self, IntArrayRef blocksize) {
 }
 
 Tensor dense_to_sparse_bsc(const Tensor& self, IntArrayRef blocksize) {
-  AT_ERROR(
-      "Conversion from ", self.layout(), " to SparseBsc is currently not supported.");
-  return self;
+  TORCH_CHECK(
+      blocksize[0] > 0 && blocksize[1] > 0,
+      "blocksize needs to be non zero, but got ",
+      blocksize);
+  TORCH_CHECK(
+      self.size(-2) % blocksize[0] == 0,
+      "Tensor size(-2) ",
+      self.size(-2),
+      " needs to be divisible by blocksize[0] ",
+      blocksize[0]);
+  TORCH_CHECK(
+      self.size(-1) % blocksize[1] == 0,
+      "Tensor size(-1) ",
+      self.size(-1),
+      " needs to be divisible by blocksize[1] ",
+      blocksize[1]);
+  auto n_batch_dim = self.dim() - 2;
+  auto is_batched = n_batch_dim > 0;
+  auto values = _batch_tile_tensor(self, blocksize);
+  auto not_zero_mask = _batch_tile_tensor((self != 0), blocksize);
+  not_zero_mask = not_zero_mask.any(-1).any(-1);
```

Comment:
Something non-trivial is going on in here... I also wonder whether double `any` could be eliminated and performed only once on values and then retiled with the block size `(blocksize[0], 1)` or something along these lines...

Reply:
NOTE: `_batch_tile_tensor` is not very efficient. If the last two dims are "contiguous", we can just reshape them directly without calling into `transpose + contiguous`.

Reply:
Yes that makes sense!

regarding the nit: I like keeping the name of the mask as it is to maintain consistency between (B/C)S(R/C) implementations. Once we have or-reduced the mask over the block shape the rest of the conversion proceeds identically to the non-blocked format.